MovieApp/MovieCategoriesViewController.swift

```diff
@@ -0,0 +1,266 @@
+        }
+        
+        for trendingMovie in movieDetails.trendingMovies{
+            trendingCollectionView.register(MovieCategoriesViewCell.self, forCellWithReuseIdentifier: trendingMovie.name)
+        }
+        
+    }
+    
+    override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
+        return .portrait
+    }
+
+    override var shouldAutorotate: Bool {
+        return true
+    }
+    
+}
+
+
+class MovieCategoriesViewCell: UICollectionViewCell {
+    
+    let thumbnailImageView = UIImageView()
+    let favoriteButton = UIButton()
+    
+    override init (frame: CGRect) {
+        super.init(frame: frame)
+
+        
+        contentView.addSubview(thumbnailImageView)
+        contentView.addSubview(favoriteButton)
+        contentView.backgroundColor = .red
+        contentView.layer.cornerRadius = 15
+        contentView.layer.masksToBounds = true
+    
+        thumbnailImageView.autoPinEdgesToSuperviewEdges()
+        thumbnailImageView.contentMode = .scaleAspectFill
+        thumbnailImageView.clipsToBounds = true
+        
+        favoriteButton.autoPinEdge(.leading, to: .leading, of: contentView, withOffset: 10)
+        favoriteButton.autoPinEdge(.top, to: .top, of: contentView, withOffset: 10)
+        favoriteButton.autoSetDimension(.width, toSize: 30)
+        favoriteButton.autoSetDimension(.height, toSize: 30)
+        favoriteButton.backgroundColor = .darkGray
+        favoriteButton.layer.cornerRadius = 15
+        favoriteButton.layer.opacity = 0.8
+        let homeSymbolConfiguration = UIImage.SymbolConfiguration(pointSize: 17, weight: .regular)
+        let homeImage = UIImage(systemName: "heart", withConfiguration: homeSymbolConfiguration)
+        favoriteButton.setImage(homeImage, for: .normal)
+
+        contentView.layer.cornerRadius = 12
+
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+extension MovieCategoriesViewController: UICollectionViewDelegate {
+
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        print(indexPath.row + 1)
+    }
+}
+
+class PopularCollectionViewDataSource: NSObject, UICollectionViewDataSource {
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let movieDetails = MovieUseCase()
+        let movie = movieDetails.popularMovies[indexPath.row]
+
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: movie.name, for: indexPath) as! MovieCategoriesViewCell
+
+        
+        cell.thumbnailImageView.load(url: URL(string: movieDetails.popularMovies[indexPath.row].imageUrl)!)
+        
+        return cell
+    }
+    
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        let movieInfo = MovieUseCase()
+        return movieInfo.popularMovies.count
```

Comment:
You can make this a oneliner.
```suggestion
        MovieUseCase().popularMovies.count
```